x-pack/plugins/security_solution/public/timelines/components/timeline/data_providers/index.tsx

```diff
@@ -96,28 +136,62 @@ export const DataProviders = React.memo<Props>(({ timelineId }) => {
   );
   const droppableId = useMemo(() => getDroppableId(timelineId), [timelineId]);
 
+  const kqlMode = useDeepEqualSelector(
+    (state) => (getTimeline(state, timelineId) ?? timelineDefaults).kqlMode
+  );
+
+  const handleChange = useCallback(
+    (mode: KqlMode) => {
+      dispatch(updateKqlMode({ id: timelineId, kqlMode: mode }));
+    },
+    [timelineId, dispatch]
+  );
+
   return (
-    <DropTargetDataProvidersContainer
-      aria-label={i18n.QUERY_AREA_ARIA_LABEL}
-      className="drop-target-data-providers-container"
-    >
-      <DropTargetDataProviders
-        className="drop-target-data-providers"
-        data-test-subj="dataProviders"
+    <>
+      <SearchOrFilterGlobalStyle />
+      <DropTargetDataProvidersContainer
+        aria-label={i18n.QUERY_AREA_ARIA_LABEL}
+        className="drop-target-data-providers-container"
       >
-        {dataProviders != null && dataProviders.length ? (
-          <Providers
-            browserFields={browserFields}
-            timelineId={timelineId}
-            dataProviders={dataProviders}
-          />
-        ) : (
-          <DroppableWrapper isDropDisabled={isLoading} droppableId={droppableId}>
-            <Empty browserFields={browserFields} timelineId={timelineId} />
-          </DroppableWrapper>
-        )}
-      </DropTargetDataProviders>
-    </DropTargetDataProvidersContainer>
+        <CustomTooltipDiv>
+          <EuiToolTip
+            className="timeline-select-search-filter-tooltip"
+            content={i18n.FILTER_OR_SEARCH_WITH_KQL}
+          >
+            <EuiSuperSelect
+              data-test-subj="timeline-select-search-or-filter"
+              hasDividers={true}
+              itemLayoutAlign="top"
+              itemClassName={timelineSelectModeItemsClassName}
+              onChange={handleChange}
+              options={options}
+              popoverProps={{
```

Comment:
Same thing here. Pull the object out of the component so it doesn't cause tooltip re-renders when the object reference changes

Reply:
This is done.